edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/SystemInfo.cs

```diff
@@ -11,7 +11,7 @@ namespace Microsoft.Azure.Devices.Edge.Agent.Core
 
     public class SystemInfo
     {
-        public SystemInfo(string operatingSystemType, string architecture, string version, ProvisioningInfo provisioning, string serverVersion, string kernelVersion, string operatingSystem, int numCpus, string virtualized, IReadOnlyDictionary<string, object> additionalProperties)
+        public SystemInfo(string operatingSystemType, string architecture, string version, ProvisioningInfo provisioning, string serverVersion, string kernelVersion, string operatingSystem, int numCpus, int totalMemory, string virtualized, IReadOnlyDictionary<string, object> additionalProperties)
```

Comment:
MaxValue for an `int` (which IIRC aliases to Int32) is 2,147,483,647.  So... would we potentially have overflow on machines with more memory?

Reply:
Great catch, thank you. Fixed in 41aa35174cc476af37b8a174a1b051ffbc764a48.